app/i18n/index.js

```diff
@@ -12,27 +25,70 @@ import ko from 'assets/i18n/ko.json';
 import nl from 'assets/i18n/nl.json';
 import pl from 'assets/i18n/pl.json';
 import ptBR from 'assets/i18n/pt-BR.json';
+import tr from 'assets/i18n/tr.json';
 import ru from 'assets/i18n/ru.json';
 import zhCN from 'assets/i18n/zh-CN.json';
 import zhTW from 'assets/i18n/zh-TW.json';
 
 const DEFAULT_LOCALE = 'en';
 
 const TRANSLATIONS = {
-    de,
-    en,
-    es,
-    fr,
-    ja,
-    ko,
-    nl,
-    pl,
-    ptBR,
-    ru,
-    zhCN,
-    zhTW
+    de: {
+        data: deLocaleData,
+        messages: de
+    },
+    en: {
+        data: enLocaleData,
+        messages: en
+    },
+    es: {
+        data: esLocaleData,
+        messages: es
+    },
+    fr: {
+        data: frLocaleData,
+        messages: fr
+    },
+    ja: {
+        data: jaLocaleData,
+        messages: ja
+    },
+    ko: {
+        data: koLocaleData,
+        messages: ko
+    },
+    nl: {
+        data: nlLocaleData,
+        messages: nl
+    },
+    pl: {
+        data: plLocaleData,
+        messages: pl
+    },
+    'pt-BR': {
+        data: ptLocaleData,
+        messages: ptBR
+    },
+    tr: {
+        data: trLocaleData,
+        messages: tr
+    },
+    ru: {
+        data: ruLocaleData,
+        messages: ru
+    },
+    'zh-CN': {
+        data: zhLocaleData,
+        messages: zhCN
+    },
+    'zh-TW': {
+        data: zhLocaleData,
+        messages: zhTW
+    }
 };
 
 export function getTranslations(locale) {
-    return TRANSLATIONS[locale] || TRANSLATIONS[DEFAULT_LOCALE];
+    const translations = TRANSLATIONS[locale] || TRANSLATIONS[DEFAULT_LOCALE];
+    addLocaleData(translations.data);
```

Comment:
hmmm, yup.. I can change this to load all the all the locales at once in an array